backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/aggregation/impl/AbstractLuceneMetricNumericFieldAggregation.java

```diff
@@ -187,8 +163,22 @@ public Builder(AbstractLuceneNumericFieldCodec<F, E> codec, LuceneSearchIndexSco
 		}
 
 		@Override
-		public LuceneMetricNumericFieldAggregation<F, E, K> build() {
-			return new LuceneMetricNumericFieldAggregation<>( this );
+		public AbstractLuceneMetricNumericFieldAggregation<F, E, K> build() {
+			if ( "sum".equals( operation ) ) {
+				return new LuceneSumNumericFieldAggregation<>( this );
+			}
+			else if ( "min".equals( operation ) ) {
+				return new LuceneMinNumericFieldAggregation<>( this );
+			}
+			else if ( "max".equals( operation ) ) {
+				return new LuceneMaxNumericFieldAggregation<>( this );
+			}
+			else if ( "avg".equals( operation ) ) {
+				return new LuceneAvgNumericFieldAggregation<>( this );
+			}
+			else {
+				throw new AssertionFailure( "Aggregation operation not supported: " + operation );
+			}
```

Comment:
I think that we can do a step further and push also the builder in each `*NumericFieldAddregation`, but I guess that if we really want to remove the if-else chain, we need to push also the `TypeSelector` and the `Factory` in each `*NumericFieldAddregation`. I'm trying to do a different commit. I'm not sure the overall code is nicer. Usually I agree it is better to avoid if and use subclasses, but maybe here the use case is quite complex, since we have for each case sub classes like: a factory, a type selector and the builder. Anyway, let's try to do that.

Reply:
Another option would be to introduce a method to create the factory, passing the codec and the method reference to each `*NumericFieldAddregation` constructor... I don't know what you think about it

Reply:
I tried!